tests/introspection/ApiSignatureTest.cs

```diff
@@ -865,5 +865,80 @@ protected virtual void CheckManagedMemberSignatures (MethodBase m, Type t, ref i
 					ReportError ($"`{t.Name}.{m.Name}` return type `{rt.Name}` is a concrete type `[Model]` and not an interface `[Protocol]`");
 			}
 		}
+
+		static bool IsDiscouraged (MemberInfo mi)
+		{
+			foreach (var ca in mi.GetCustomAttributes ()) {
+				switch (ca.GetType ().Name) {
+				case "ObsoleteAttribute":
+				case "ObsoletedAttribute":
+				case "DeprecatedAttribute":
+					return true;
+				}
+			}
+			return false;
+		}
+
+		[Test]
+		public void AsyncCandidates ()
+		{
+			int n = 0;
+			Errors = 0;
+			ErrorData.Clear ();
+
+			foreach (Type t in Assembly.GetTypes ()) {
+
+				if (!NSObjectType.IsAssignableFrom (t))
+					continue;
+
+				if (t.GetCustomAttribute<ProtocolAttribute> () != null)
+					continue;
+				if (t.GetCustomAttribute<ModelAttribute> () != null)
+					continue;
+
+				// let's not encourage the use of some API
+				if (IsDiscouraged (t))
+					continue;
+
+				CurrentType = t;
+
+				var methods = t.GetMethods (Flags);
+				foreach (MethodInfo m in methods) {
+					if (m.DeclaringType != t)
+						continue;
+
+					// skip properties / events
+					if (m.IsSpecialName)
+						continue;
+
+					// some calls are "natively" async
+					if (m.Name.IndexOf ("Async", StringComparison.Ordinal) != -1)
+						continue;
+
+					// let's not encourage the use of some API
+					if (IsDiscouraged (m))
+						continue;
+
+					// is it a candidate ?
+					var p = m.GetParameters ();
+					if (p.Length == 0)
+						continue;
+					var last = p [p.Length - 1];
+					if (last.Name != "completionHandler")
```

Comment:
It looks like we have a lot of candidate that didn't name their completion argument `completionHandler`,, just something similar: https://gist.github.com/rolfbjarne/35cf6852ed2cbbed7d241a347ffddda7